src/vtcomposite.cpp

```diff
@@ -692,104 +716,183 @@ struct LocalizeWorker : Napi::AsyncWorker
+                            if (idx < class_key_idx)
                             {
+                                class_key_idx = idx;
                                 class_value = property.value();
                             }
-                            continue;
+                            // wait till we are done looping through all properties before we add class value to final_properties
                         }
 
-                        // preserve original params.language_property value
-                        if (property_key == baton_data_->language_property)
+                        else if (
+                            utils::startswith(property_key, baton_data_->language_property) ||
+                            utils::startswith(property_key, baton_data_->language_prefix + baton_data_->language_property))
                         {
-                            name_value = property.value();
+                            // check if the property is of higher precedence that class key encountered so far
+                            std::uint32_t idx = static_cast<std::uint32_t>(std::find(language_key_precedence.begin(), language_key_precedence.end(), property_key) - language_key_precedence.begin());
+                            if (idx < language_key_idx)
+                            {
+                                language_key_idx = idx;
+                                language_value = property.value();
+                            }
 
-                            // if no language was specified, we want the name value to be constant
-                            if (baton_data_->language.empty())
+                            // preserve original language value and wait till finish looping through all properties to assign a value
+                            if (property_key == baton_data_->language_property)
                             {
-                                properties.emplace_back(baton_data_->language_property, property.value());
-                                name_was_set = true;
+                                original_language_value = property.value();
+                            }
+                            else
+                            {
+                                if (keep_every_language)
+                                {
+                                    if (!utils::startswith(property_key, baton_data_->language_prefix))
+                                    {
+                                        final_properties.emplace_back(property_key, property.value());
+                                    }
+                                    // else – drop properties that start with a prefix
+                                }
+                                // else – wait till we are done looping through all properties to add class value to final_properties
```

Comment:
this says class value - do you mean language value?

Reply:
Fixed in 3711e05918f95ff3494302a0c36e07fe1bdfad08.